core/primitives-core/src/account.rs

```diff
@@ -112,20 +112,19 @@ struct LegacyAccount {
 
 impl BorshDeserialize for Account {
     fn deserialize(buf: &mut &[u8]) -> Result<Self, io::Error> {
-        if buf.len() == std::mem::size_of::<LegacyAccount>() {
```

Comment:
I'd just hard-coded the constant here, deserializing LegacyAccount every time feels like it can affect performance negatively. We can than have a test which compares the constant with the size of serialized account.

Reply:
Ah yes, you are completely right!